spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -430,6 +432,20 @@ public void setFileExistsMode(FileExistsMode fileExistsMode) {
 		}
 	}
 
+	public void setChmodOctal(String chmod) {
+		Assert.notNull(chmod, "'chmod' cannot be null");
+		this.chmod = Integer.parseInt(chmod, 8);
+	}
+
+	public void setChmodDecimal(int chmod) {
+		Assert.isTrue(isChmodCapable(), "chmod operations not supported");
```

Comment:
Looks like this `Assert` has to be present in the other `setChmod...` as well.
Plus both of them are would be better with JavaDocs with `@since 4.3`.